### PR TITLE
[MNT] remove redundant soft dependency checks

### PR DESCRIPTION
**Pull request by CodeLionX**
#### Reference Issues/PRs

none, but inspired by [this PR comment](https://github.com/aeon-toolkit/aeon/pull/2091#discussion_r1775678022)

#### What does this implement/fix? Explain your changes.

- removes redundant calls to `_check_soft_dependencies` in estimators (this is already done by the `__init__`-function of the base classes
- adds the correct version bound for `pyod` in `PyODAdapter` since a new version with the `pyod.__version__`-attribute was published

#### Does your contribution introduce a new dependency? If yes, which one?

no

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

### CODE REVIEW

File: aeon/clustering/_k_shapes.py
Comment (MatthewMiddlehurst) [inline aeon/clustering/_k_shapes.py:1]: good reminder to remove this, not here ofc 🙂

File: aeon/anomaly_detection/_pyodadapter.py [hunk 1]
```diff
@@ -81,10 +81,7 @@ class PyODAdapter(BaseAnomalyDetector):
         "capability:univariate": True,
         "capability:missing_values": False,
         "fit_is_empty": False,
-        # Omit the version specification until PyOD has __version__
-        # (https://github.com/yzhao062/pyod/pull/584 in dev but not released yet)
-        # "python_dependencies": ["pyod>=1.1.3"]
-        "python_dependencies": ["pyod"],
+        "python_dependencies": ["pyod>=2.0.1"],
```
Comment (MatthewMiddlehurst) [inline aeon/anomaly_detection/_pyodadapter.py:84]: is the `2.0.1` needed? better to leave if it not.

**Review by MatthewMiddlehurst — APPROVED** [review]
Have of question on `pyod` but LGTM if you want to keep it as is.

File: aeon/anomaly_detection/_pyodadapter.py (hunk 1 above)
Comment (CodeLionX) [reply aeon/anomaly_detection/_pyodadapter.py:84]: No, but at least `1.1.3`. Starting with `2.0.1`, the `__version__`-attribute is there.

Comment (MatthewMiddlehurst) [reply aeon/anomaly_detection/_pyodadapter.py:84]: May be better to remove the bound here and add it to `pyproject.toml` don't think there are any other usages which would require a lower bound?

Comment (CodeLionX) [reply aeon/anomaly_detection/_pyodadapter.py:84]: I understand 👍🏼

**Review by TonyBagnall — APPROVED** [review]
lgtm

**Pull request merged**